internal/iptables/iptables.go

```diff
@@ -0,0 +1,80 @@
+package iptables
+
+import (
+	"errors"
+	"fmt"
+	"net/netip"
+	"strconv"
+
+	"github.com/matheuscscp/gke-metadata-server/api"
+	"k8s.io/kubernetes/pkg/util/iptables"
+	"k8s.io/utils/exec"
+)
+
+func LoadAndAttach(emulatorAddr netip.Addr, emulatorPort int) func() (func() error, error) {
+	return func() (func() error, error) {
+		// Create the following iptables rules:
+		// iptables -t nat -A OUTPUT -d 169.254.169.254 -p tcp --dport 80 -j DNAT --to-destination <emulatorAddr>
+		// iptables -A FORWARD -d <emulatorIP> -p tcp --dport <emulatorPort> -m state --state NEW,ESTABLISHED,RELATED -j ACCEPT
+
+		// This rule essentially rewrites the destination of packets targeting the
+		// GKE metadata server with the ip:port address of the emulator, i.e. it
+		// effectively modifies the destination fields of matching packets.
+		ipTables := iptables.New(exec.New(), iptables.ProtocolIPv4)
+
+		// match conditions
+		natTableArgs := [...]string{
+			"-d", api.GKEMetadataServerAddressDefault,
+			"-p", "tcp",
+			"--dport", strconv.Itoa(api.GKEMetadataServerPortDefault),
+
+			// action taken
+			"-j", "DNAT",
+			"--to-destination", emulatorAddr.String(),
+		}
+
+		filterTableArgs := [...]string{
+			"-d", emulatorAddr.String(),
+			"-p", "tcp",
+			"--dport", strconv.Itoa(emulatorPort),
+			"-m", "state", "--state", "NEW,ESTABLISHED,RELATED", // new or established connections
+			// action taken
+			"-j", "ACCEPT",
+		}
+
+		close := func() error {
+			err1 := ipTables.DeleteRule(
+				iptables.Table(iptables.TableNAT),
+				iptables.ChainOutput,
+				natTableArgs[:]...,
```

Comment:
I don't think you need `[:]` here either, just `natTableArgs...`

Reply:
I tried doing `natTableArgs...` and it didn't work

Reply:
Okay I understand now, it's because `[...]type{}` initializes an array, which has fixed-size, and not a slice. You only need these `[:]` when the symbol is an array, so if you remove the ellipsis from the declaration above you can also remove this here